main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+)
+
+func readCsv(pokemonId string) ([]string, error) {
+	fileName := "pokemon.csv"
+
+	fs, err := os.Open(fileName)
+
+	if err != nil {
+		return nil, fmt.Errorf("can not open the file, err is %+v", err)
+	}
+
+	defer fs.Close()
+
+	reader := csv.NewReader(fs)
+
+	for {
+		content, err := reader.Read()
+
+		if err != nil && err != io.EOF {
+			return nil, fmt.Errorf("can not read, err is %+v", err)
+		}
+		if err == io.EOF {
+			break
+		}
+
+		if content[0] == pokemonId {
+			return content, nil
+		}
+	}
+
+	return nil, fmt.Errorf("Not found, err is %+v", err)
+}
+
+func handler(response http.ResponseWriter, request *http.Request) {
+	keys, ok := request.URL.Query()["id"]
+
+	if !ok || len(keys[0]) < 1 {
+		log.Fatal("Url Param 'id' is missing")
+		return
+	}
+
+	key := keys[0]
+
+	content, errCsv := readCsv(key)
+
+	if errCsv != nil {
+		log.Fatal(fmt.Errorf("can not read the csv file: %v", errCsv))
+	}
+
+	jsonEncoder := json.NewEncoder(response)
+	jsonEncoder.SetIndent("", "  ")
+
+	err := jsonEncoder.Encode(content)
+
+	if err != nil {
+		log.Fatal(fmt.Errorf("impossible to encode pokemons: %v", err))
```

Comment:
I consider this error is not a `fatal` one because it occurred in the request handling, but if this fails the API could be kept running attending to other requests that may not fail, so you can log the error and return an `internal server error` status with the error message to let the user know that something fails and can decide if retry, skip, or something. 
I consider this error is not a `fatal` one because it occurred in the request handling, if this fails the API could be kept running attending to other requests that may not fail, so you can log the error and return an `internal server error` status with the error message to let the user know that something fails and can decide if retry, skip, or something.

Reply:
done